controller/engine_controller.go

```diff
@@ -911,18 +911,6 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 	if cliAPIVersion >= engineapi.MinCLIVersion {
 		// Cannot continue to start restoration if expansion is not complete
 		if engine.Spec.VolumeSize > engine.Status.CurrentSize {
-			// Cannot tolerate the rebuilding replicas during the expansion
-			hasRebuildingReplicas := false
-			for name, mode := range engine.Status.ReplicaModeMap {
-				if mode == longhorn.ReplicaModeWO {
-					engine.Status.ReplicaModeMap[name] = longhorn.ReplicaModeERR
-					hasRebuildingReplicas = true
-				}
-			}
-			if hasRebuildingReplicas {
-				m.logger.Warn("Cannot contain any rebuilding replicas during engine expansion, will mark all rebuilding replicas as ERROR")
-				return nil
-			}
```

Comment:
Why can we remove this logic?

Reply:
Previously the expansion is offline and should be fast, hence waiting for the rebuilding complete is not user-friendly. Now we can have the online expansion, which does not care if rebuilding exists or not. Maybe we need to continue disabling the rebuilding for offline expansion. But we can do it after your volume controller refactoring merge.

Reply:
I see. 

But will there be a problem if we try to expand a replica when it is being rebuilt?

Reply:
Based on the integration test results, No.
For the replicas, the expansion is like snapshotting with some extra operations. If operation snapshotting works fine with rebuilding, we don't need to worry about the expansion part.

Reply:
Sounds good. Thank you for the confirmation